### ISSUE TITLE
Softer blocking for temporary, i.e. non-permanent block should left existing comments alone

### ISSUE THREAD

**Issue by umputun** [umputun/remark#377]
Currently, the temporary block doesn't remove user's comment but hides them for a given time and prevents the user from posting more. Practically, this logic seems to be too restrictive because the reason for such a temporary ban is usually a heated discussion or some other c concrete topic/conversation triggered the ban. However, it doesn't mean we have to hide/remove all previous comment from the user as they can be fine and legit.

The proposed change will not hide such comments, but won't allow the user to post more for the duration of the ban.

**Comment by umputun** [comment]
@Reeywhaar - I think there is a bug on UI side rendering blocked users. I see the `blocked` status returned by REST but UI seems to ignore it in some cases. It is not always - the first time (after I set the blocking status) it shows comments faded out with "blocked" badge, however page refresh shows them as a regular user/comment.It also looks like Show/Hide settings "restores" it till the next refresh.

_in order to reproduce it log in as admin (dev_user) block some other user temporary and refresh the page._

**Comment by Reeywhaar** [comment]
Hm, can't reproduce it. What I see:
If I block someone comments get marked as "blocked"

![image](https://user-images.githubusercontent.com/884649/61663511-71e19280-acd9-11e9-879d-4e233dced1b1.png)

After refresh badge changes to "deleted". *which is a bug, block should take precedence over delete*

If I enter as non-admin comment looks like this:
![image](https://user-images.githubusercontent.com/884649/61663602-a81f1200-acd9-11e9-91e3-87608f96f629.png)

I don't see any bug with users been disappearing from preferences, despite I tried to block for a month/week/day. Also checked in Safari.

Upon opening preferences, there is request being made to `/api/v1/admin/blocked?site=remark` what does it look like when issue present?

Am I understand right that there is checkers behavior: blocking someone (looks good) -> refresh (looks bad) -> refresh (looks good)?

> I see the `blocked`
you mean that comment has `.user.block === true`? Or `.delete === true`? Or both?

**Comment by umputun** [comment]
> blocking someone (looks good) -> refresh (looks bad) -> refresh (looks good)?

not exactly.  blocking someone (looks good) -> refresh (looks bad) -> refresh (looks bad) -> more refreshes (still bad).

> you mean that comment has .user.block === true? Or .delete === true? Or both?

 .user.block === true only. The current master doesn't set `.delete` anymore (https://github.com/umputun/remark/commit/a60072564e16384bf33ab7d1fa566ba13ab1e2fd) for any temporarry bans, but for permanent only.

**Comment by umputun** [comment]
this is a small video demoing what I've seen

http://4dropz.com/qJR8maCoqARc

**Comment by umputun** [comment]
As you see on the video - refresh breaks blocking rendering but going back to settings restores it till the next refresh

**Comment by Reeywhaar** [comment]
Found the problem, solution is on the way. But found another issue: last comments endpoint doesn't trim blocked users. I think it should?

**Comment by umputun** [comment]
Do you mean temporary or permanent block? Temporary shouldn't trim anymore, only permanent should do it.

**Comment by Reeywhaar** [comment]
Temporary, I think it's confusing that last comments have blocked user rendered fine.

**Comment by Reeywhaar** [comment]
![Screenshot 2019-07-23 at 01 30 02](https://user-images.githubusercontent.com/884649/61669662-8cbc0300-ace9-11e9-9cf8-3075d2d6dada.png)

**Comment by umputun** [comment]
I see what you mean - we don't do faded thingy for last comment widget. Probably you right, I'll make the change to filter them out from the last comments widget.

Regrading the picture - is it you or me doing "this user was blocked"? Somehow I have expected to see the original text here unless comment marked as deleted

**Comment by Reeywhaar** [comment]
"This user was blocked" will be shown to non-admin users, as well as avatar will be ghost. Also I think to remove time on the right, and deny voting for comment

**Comment by umputun** [comment]
just to clarify - the goal of this ticket is **not to delete** (and not to hide) comments for temporarily blocked user, but make blocking status visible (color and blocked badge) and not allow them to interact with the system till block expired

**Comment by Reeywhaar** [comment]
Oh, ok, overdone it.

**Comment by umputun** [comment]
This is the practical use case I have witnessed multiple times - a user acts like a normal human being until some "hot topic" mentioned. From this point, he started to post offensive, inappropriate, and silly comments.  Before the proposed change, my actions were limited. I don't want to remove/hide his innocent comments, but want to stop the madness and (maybe) delete a few recent comments. However, the only action I was able to do was a block (hiding all comments) or manual removal (allowing him to continue).

With this change, we will able to suspend such undesired activity without extream punishment. 

I hope it makes sense.

**Comment by Reeywhaar** [comment]
Ok, understand, so I think there is no necessity to trim last comments too

**Comment by umputun** [comment]
it works, thank you. One minor issue/inconsistency - blocking action refreshes whatever part of the screen it needs to update, and admin may see the effect immediately. Attempt to unblock **right away**  refreshes appropriately as well, however, if between the block and unblock actions admin does any refresh, unblocking works "quietly", i.e. without updating/refreshing the screen. Pls, note - I meant unblock by clicking a link under the user name, not on settings view.

Obviously, this is not a big deal, but it will be helpful to make it right. Let me know if you want a video demonstrating it.

**Comment by umputun** [comment]
LGTM
